### PR TITLE
Implement #5253(Extended vein Mining)

### PR DESCRIPTION
**Pull request by ToMe25**
## Changes proposed in this pull request:
* Added a key to extend vein mining to mine everything(#5253)
* Added a Config Option to set the Max Vein Mined Blocks

### CODE REVIEW

File: src/main/resources/assets/mekanism/lang/de_de.lang [hunk 1]
```diff
@@ -385,9 +385,11 @@ upgrade.muffling=Dämpfung
 upgrade.muffling.desc=Verringert den von Maschinen produzierten Schall.
 
 # Key description text
-mekanism.key.mode=Moduswechsel
-mekanism.key.armorMode=Küstungsmoduswechsel
+mekanism.key.mode=Item Modus wechsel
+mekanism.key.armorMode=Rüstungs Modus wechsel
+mekanism.key.armorMode=Fuß Modus wechsel
```
Comment (pupnewfster) [inline src/main/resources/assets/mekanism/lang/de_de.lang]: You have the same lang key `mekanism.key.armorMode` here twice

Comment (ToMe25) [reply src/main/resources/assets/mekanism/lang/de_de.lang]: oops i guess i know how that got there...

File: src/main/java/mekanism/client/MekKeyHandler.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ public static boolean getIsKeyPressed(KeyBinding keyBinding) {
     public void keyTick() {
         for (int i = 0; i < keyBindings.length; i++) {
             KeyBinding keyBinding = keyBindings[i];
-            boolean state = keyBinding.isPressed();
+            boolean state = keyBinding.isKeyDown();
```
[the remark this answers is not in the document]
Comment (thiakil) [reply src/main/java/mekanism/client/MekKeyHandler.java]: "to fix the key sending" - please explain

**Comment by thiakil** [comment]
_Please_ don't combine multiple features in the one PR, it makes it a harder to review, and changes requested in one part hold back all the others.

**Comment by ToMe25** [comment]
OK than i will move the Coal Gasification to a new PR.

**Comment by ToMe25** [comment]
OK I finished Splitting the Coal Gasification in a new PR(#5547).

File: src/main/java/mekanism/client/MekKeyHandler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (thiakil) [reply src/main/java/mekanism/client/MekKeyHandler.java]: @ToMe25

[the remark this answers is not in the document]
Comment (ToMe25) [reply src/main/java/mekanism/client/MekKeyHandler.java]: If i don't change that but keep everything else like this it keeps extended vein mining active for ever and never(only after the next client restart) disables it after releasing the Key.
And sometimes(in some very rare cases) disables it if you press the key just very short time.

[the remark this answers is not in the document]
Comment (ToMe25) [reply src/main/java/mekanism/client/MekKeyHandler.java]: sorry wasn't seeing your Comment :)

[the remark this answers is not in the document]
Comment (thiakil) [reply src/main/java/mekanism/client/MekKeyHandler.java]: javadoc points to `isKeyDown` being correct

File: src/main/java/mekanism/common/item/ItemAtomicDisassembler.java [hunk 3]
```diff
@@ -348,15 +353,21 @@ public boolean canSend(ItemStack itemStack) {
         private final RayTraceResult rayTraceResult;
         private final Block startBlock;
         private final boolean isWood;
+        private final int maxRange;
 
         public Finder(EntityPlayer p, ItemStack s, Coord4D loc, RayTraceResult traceResult) {
+            this(p, s, loc, traceResult, -1);
```
Comment (thiakil) [inline src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: Is this constructor used anywhere? No point having it if not

File: src/main/java/mekanism/common/item/ItemAtomicDisassembler.java [hunk 4]
```diff
@@ -128,8 +130,9 @@ public boolean onBlockStartBreak(ItemStack itemstack, BlockPos pos, EntityPlayer
                     isOre = true;
                 }
             }
-            if (getMode(itemstack) == 3 && isOre && !player.capabilities.isCreativeMode) {
-                Set<Coord4D> found = new Finder(player, stack, new Coord4D(pos, player.world), raytrace).calc();
+            boolean extended = Mekanism.keyMap.has(player, KeySync.EXTENDEDMINING);
+            if (getMode(itemstack) == 3 && (isOre && !player.capabilities.isCreativeMode || extended)) {
+                Set<Coord4D> found = new Finder(player, stack, new Coord4D(pos, player.world), raytrace, extended ? 10 : -1).calc();
```
Comment (thiakil) [inline src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: This radius needs to be added as a config option, and an option to enable the extended vein mining to begin with

File: src/main/java/mekanism/common/item/ItemAtomicDisassembler.java (hunk 3 above)
Comment (ToMe25) [reply src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: thats just the old constructor thats why i didn't removed it, i will remove it than.

**Comment by ToMe25** [comment]
Yes i have tested it with veinminer and oreexcavation. no problems

File: src/main/java/mekanism/common/item/ItemAtomicDisassembler.java (hunk 4 above)
Comment (ToMe25) [reply src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: OK did that

File: src/main/java/mekanism/common/item/ItemAtomicDisassembler.java (hunk 3 above)
Comment (ToMe25) [reply src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: ok removed it.

File: src/main/java/mekanism/common/item/ItemAtomicDisassembler.java [hunk 5]
```diff
@@ -368,10 +366,11 @@ public Finder(EntityPlayer p, ItemStack s, Coord4D loc, RayTraceResult traceResu
             rayTraceResult = traceResult;
             isWood = OreDictCache.getOreDictName(stack).contains("logWood");
             maxRange = range;
+            maxCount = MekanismConfig.current().general.DISASSEMBLER_COUNT.val() - 1;
```
Comment (pupnewfster) [inline src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: With the -1 doesn't this change the amount that it actually can do? Or is this fixing that it was previously doing 129 before (haven't looked too closely at the code yet)

Comment (ToMe25) [reply src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: it fixes mining one more than set in the config.

**Comment by ToMe25** [comment]
I hope i merged everything right.

File: src/main/java/mekanism/common/item/ItemAtomicDisassembler.java [hunk 6]
```diff
@@ -144,7 +148,9 @@ public boolean onBlockStartBreak(ItemStack itemstack, BlockPos pos, EntityPlayer
                     player.world.playEvent(WorldEvents.BREAK_BLOCK_EFFECTS, coord.getPos(), Block.getStateId(state));
                     player.world.setBlockToAir(coord.getPos());
                     block2.breakBlock(player.world, coord.getPos(), state);
-                    block2.dropBlockAsItem(player.world, coord.getPos(), state, 0);
+                    if(!player.capabilities.isCreativeMode) {
```
Comment (pupnewfster) [inline src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: The player can't be in creative here as it is checked at the top of the method.

Comment (ToMe25) [reply src/main/java/mekanism/common/item/ItemAtomicDisassembler.java]: oh ok ill change that than

**Comment by pupnewfster** [comment]
Cleaned up a couple typos in the config options and a bit of spacing stuff. For the most part everything seems to work fine. Going to talk with @thiakil some and see what his thoughts on it are. Initially I had mentioned to him that I think we should potentially hold off and make it some sort of upgrade like in #3904 . Though upon thinking about it more I am not really against this existing in this form before work on v10 ends up happening.

**Comment by pupnewfster** [comment]
Cleaned up a couple typos in the config options and a bit of spacing stuff. For the most part everything seems to work fine. Going to talk with @thiakil some and see what his thoughts on it are. Initially I had mentioned to him that I think we should potentially hold off and make it some sort of upgrade like in #3904 . Though upon thinking about it more I am not really against this existing in this form before work on v10 ends up happening.

**Comment by ToMe25** [comment]
I didn't remembered about it at the time i merged, but i made it intentionally a way that allows extended vein miner for players that are in creative mode and just doesn't drops blocks.
Should i change it back to that or leave it like it is now?

**Comment by pupnewfster** [comment]
I think leaving it how it is now in terms of the creative thing is fine. One thing we have discussed some is instead of having them have to hold another button is to just add the "extended" vein mining as another mode. I may look into doing this later today before the 9.8.0 release or you can attempt to do it if you want. It should be *relatively* easy to add a new mode now, after the cleanup I did a few days ago to the whole class.

**Comment by pupnewfster** [comment]
Going through and making it as another mode. Also setting up a system that will allow *most* modes to be disabled and then cycling through will just skip over it if it is not enabled.

**Pull request merged**